hack/ovn-kind-cno.sh

```diff
@@ -0,0 +1,200 @@
+  echo "Copying kubeconfig"
+  docker cp ~/admin.conf $n:/etc/kubernetes/kubeconfig
+  docker exec $n chmod 777 /etc/kubernetes/kubeconfig
+  echo "Modifying os-release for Multus"
+  # required for Multus platform check
+  docker exec $n sed -i 's/ID=.*/ID=rhcos/' /etc/os-release
+done
+
+# Create Proxy resource
+kubectl create -f https://raw.githubusercontent.com/openshift/api/e7fa4b871a25985ef0cc36c2fbd9f2cb4445dc9c/config/v1/0000_03_config-operator_01_proxy.crd.yaml
+
+# Create Network resource
+kubectl create -f https://raw.githubusercontent.com/openshift/api/e7fa4b871a25985ef0cc36c2fbd9f2cb4445dc9c/config/v1/0000_10_config-operator_01_network.crd.yaml
+
+# Create cluster operator
+kubectl create -f https://raw.githubusercontent.com/openshift/machine-api-operator/050a65a2bdabcc2c2f17036de967c6bcee6d6a48/config/0000_00_cluster-version-operator_01_clusteroperator.crd.yaml
+
+if [ "$BUILD_OVN" = true ] || [ "$BUILD_CNO" = true ]; then
+  pushd $CNO_TEMPLATES
+  DEPLOYMENT_TEMPLATE=$(ls 0000*deployment*)
+  if [ -z "$DEPLOYMENT_TEMPLATE" ]; then
+    echo "error locating deployment template in $CNO_TEMPLATES"
+    exit 1
+  fi
+  cp $DEPLOYMENT_TEMPLATE deployment.yaml.bk
+  if [ "$BUILD_OVN" = true ]; then
+    sed -i 's/".*origin-ovn-kubernetes:.*/"origin-ovn-kubernetes:dev"/' $DEPLOYMENT_TEMPLATE
+  fi
+  if [ "$BUILD_CNO" = true ]; then
+    sed -i "s#quay.io/openshift/origin-cluster-network-operator:.*#$CNO_IMAGE#" $DEPLOYMENT_TEMPLATE
+  fi
+fi
+
+echo "Creating CNO operator"
+for f in $(ls $CNO_TEMPLATES| grep 0000| grep -v credentials); do
+  kubectl create -f ${CNO_TEMPLATES}/$f
+done
+
+if [ "$BUILD_OVN" = true ] || [ "$BUILD_CNO" = true ]; then
+  mv deployment.yaml.bk $DEPLOYMENT_TEMPLATE
+  popd
+fi
+
+count=1
+until kubectl get pod -n openshift-network-operator -o jsonpath='{.items[0].metadata.name}' --field-selector status.phase=Running &> /dev/null ;do
+  if [ $count -gt 15 ]; then
+    echo "Network operator not running"
+    exit 1
+  fi
+  echo "Network operator pod not available yet on attempt $count"
+  count=$((count+1))
+  sleep 10
+
+done
+
+CNO_POD=$(kubectl get pod -n openshift-network-operator -o jsonpath='{.items[0].metadata.name}' --field-selector status.phase=Running)
+if [ -z "$CNO_POD" ]; then
+    echo "Cannot find running CNO pod"
+    exit 1
+fi
+
+if [ "$BUILD_CNO" != true ]; then
+  echo "WARNING: patching CNO operator pod for OVN-K8S, deployment will no longer function if this pod is restarted"
+  kubectl -n openshift-network-operator  exec $CNO_POD sed -i '/host-run-netns/{n;s/readOnly.*/mountPropagation: Bidirectional/}' /bindata/network/ovn-kubernetes/ovnkube-node.yaml > /tmp/ovnkube-node.yaml
+  kubectl cp /tmp/ovnkube-node.yaml openshift-network-operator/${CNO_POD}:/bindata/network/ovn-kubernetes/
+fi
+
+echo "Creating OVN CNO config"
+cat << EOF | kubectl create -f -
+apiVersion: config.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  clusterNetwork:
+  - cidr: 10.244.0.0/16
```

Comment:
Probably we want to make it customizable via environment variables. I'm not quite sure if this doesn't colide with some of our internal ip addressing something like:
```
CLUSTER_CIDR=${CLUSTER_CIDR:-"10.244.0.0/16"}
SERVICE_NETWORK=${SERVICE_NETWORK:-"10.96.0.0/12"}
```
[...]
```
cat << EOF | kubectl create -f -
apiVersion: config.openshift.io/v1
kind: Network
metadata:
  name: cluster
spec:
  clusterNetwork:
  - cidr: ${CLUSTER_CIDR}
    hostPrefix: 24
  networkType: OVNKubernetes
  serviceNetwork:
  - ${SERVICE_NETWORK}
EOF
```

Reply:
thanks. will fix

Reply:
> Probably we want to make it customizable via environment variables. I'm not quite sure if this doesn't colide with some of our internal ip addressing something like:

You should really use the default service range (`172.30.0.0/16`) just for recognizability.
And using a cluster CIDR in the 172.16.0.0 – 172.31.255.255 block as well ought to avoid both RH VPN and local wifi network IP ranges for most people.

Reply:
I think these are default flannel values? I'll change it to what you suggested.